### DISCUSSION TITLE
embedded control?

### DISCUSSION THREAD

**Discussion by maholli** [cariboulabs/cariboulite#21] [Q&A]
First off, I'm super excited for the CaribouLite. Thanks for all the great documentation!

For my use cases I want to command the CaribouLite board with a microcontroller and not an RPi. Now I understand that isn't the intention of the project, but it seems quite doable in firmware and would really open up the low-power applications of this awesome project. 

In practice, I'm imagining a microcontroller board supplying power & loading the bitstream while also providing an SPI/SDIO connection for the FPGA to save IQ samples. 

Is this something the Caribou Labs team is considering? I know this doesn't take advantage of all your awesome SMI work, but it does leverage the compact RF front end you've clearly put a lot of effort into. 

Thanks
Max

**Comment by a-oxide** [comment]
I am not confident this is possible looking at the current implementation. You might be better off designing your own revision of the hardware.
